awswrangler/s3/_describe.py

```diff
@@ -176,33 +113,28 @@ def describe_objects(
 
     """
     s3_client = _utils.client(service_name="s3", session=boto3_session)
-    return _describe_objects(
+
+    paths = _path2list(
         path=path,
         s3_client=s3_client,
-        version_id=version_id,
-        use_threads=use_threads,
         last_modified_begin=last_modified_begin,
         last_modified_end=last_modified_end,
         s3_additional_kwargs=s3_additional_kwargs,
     )
+    if len(paths) < 1:
+        return {}
 
-
-def _size_objects(
-    path: Union[str, List[str]],
-    s3_client: "S3Client",
-    version_id: Optional[Union[str, Dict[str, str]]] = None,
-    use_threads: Union[bool, int] = True,
-    s3_additional_kwargs: Optional[Dict[str, Any]] = None,
-) -> Dict[str, Optional[int]]:
-    desc_list = _describe_objects(
-        path=path,
-        s3_client=s3_client,
-        version_id=version_id,
-        use_threads=use_threads,
-        s3_additional_kwargs=s3_additional_kwargs,
+    executor = _get_executor(use_threads=use_threads)
+    resp_list = ray_get(
+        executor.map(
+            _describe_object,
+            s3_client,
+            paths,
+            itertools.repeat(s3_additional_kwargs),
+            [version_id.get(p) if isinstance(version_id, dict) else version_id for p in paths],
+        )
     )
-    size_dict: Dict[str, Optional[int]] = {k: d.get("ContentLength", None) for k, d in desc_list.items()}
-    return size_dict
+    return dict(resp_list)
 
 
 def size_objects(
```

Comment:
Could you also add kwargs validation, please? Since the related PR was already merged